src/server/server.py

```diff
@@ -3955,6 +4018,18 @@ def heartbeat_thread_function():
             if (heartbeat_thread_function.iter_tracker % (4*HEARTBEAT_DATA_RATE_FACTOR)) == (2*HEARTBEAT_DATA_RATE_FACTOR):
                 CLUSTER.emitStatus()
 
+            # collect vrx lock status
+            if (heartbeat_thread_function.iter_tracker % (10*HEARTBEAT_DATA_RATE_FACTOR)) == 0:
+                if vrx_controller:
+                    # if vrx_controller.has_connection
+                    vrx_controller.get_node_lock_status()
+                    vrx_controller.request_variable_status()
+
+            if (heartbeat_thread_function.iter_tracker % (10*HEARTBEAT_DATA_RATE_FACTOR)) == 4:
+                # emit display status with offset
+                if vrx_controller:
+                    emit_vrx_list()
```

Comment:
Instead of two "if vrx_controller:" statements, you can one that gates a block containing the two "if (heartbeat_thread_function.iter_tracker..." statements.

Reply:
I thought about that, but it would run the vrx check every heartbeat. Compared to that method, this one has the effect of adding extra checks to the system while vrx control is disabled—but reducing them when it's enabled (and the system has more to do otherwise).

I don't feel strongly about it, but that was the reasoning.

Reply:
A single 'vrx_controller' check would be quite fast (whereas the other does math) and would isolate that vrx_controller-related section of the code, but I don't feel strongly about it either (and I get what you're saying about every heartbeat).  Not going to make a noticeable difference either way.